### PR TITLE
[Feature] [#142] 3ème dose

### PR DESCRIPTION
**Pull request by pylapp**
## Description

Implementation of 3rd shot in the app.

## Changes / Definition of done

- [x] Remove any objects related to Chronodose (configuration, UI, wording)
- [x] Update business objects to manage daily slots
- [x] Update API to request for daily slots
- [x] Requests for data
- [x] Filter available slots by day (total is always displayed, or nothing)
- [x] Implement filtering to get vaccination centers with 3rd shot
- [x] Update GUI to display vaccination centers with 3rd shot
- [x] Make review of a11y (Voice Over, dynamic fonts)
- [x] Add SPDX headers in source files
- [x] Remove objects related to "appointment_schedules"? Seems not used anymore.
- [x] If no slots available, say it or don't display (cf. [web front](https://dev.vitemado.se/dose-rappel/))

## Related issue

_Link to the [Github issue #142](https://github.com/CovidTrackerFr/vitemadose-ios/issues/142)_
_Link to the [Github issue #150](https://github.com/CovidTrackerFr/vitemadose-ios/issues/150)_

## What I tested

- Display of booster shot in UI
- Compared results with web front

## Regression risks

_Nothing identified_

## Screenshots

|Before|After|
|-|-|
| IMG_9526|IMG_9525 |

## Checklist
- [x] I have installed SwiftLint and made sure code formatting is correct
- [x] I have performed a self-review of my own code
- [x] I tested my changes on real device
- [x] My changes generate no new warnings
- [x] I have commented my code in hard-to-understand areas
- [x] Any dependent changes have been merged into **develop**
- [x] I have checked there aren't other open [Pull Requests](https://github.com/CovidTrackerFr/vitemadose-ios/pulls) for the same update/change

### CODE REVIEW

**Comment by pylapp** [comment]
@victor-sarda Ready for review!

**Comment by rozierguillaume** [comment]
😍

**Comment by pylapp** [comment]
@victor-sarda It's ready (again) for review. I made also a refactor so as to remove a field backend wanted to delete.

File: HOW-TO.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# FAQ
+
+## Mettre en place un serveur local pour jouer avec des données
```
Comment (pylapp) [inline HOW-TO.md:3]: "Mettre en place un serveur local pour jouer avec des données **comme si renvoyées par le backend**"

File: HOW-TO.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+<html lang="fr">
+    <head>
+        <title>Vite Ma Dose (localhost)</title>
+    </head>
+    <body>
+        <h1>Youpi !</h1>
+    </body>
+</html>
+```
+5. Via un navigateur web, saisir l'URL du site web, ici **http://w.x.y.z:8888**. Un page avec le texte _Youpi !_ doit être affichée ; sinon il y a vraisemblablement un problème de configuration de votre serveur web.
```
Comment (pylapp) [inline HOW-TO.md:25]: Typo : "Une"

File: HOW-TO.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+6. Créer dans le dossier web un sous-dossier nommé _vitemadose_, au même niveau que le fichier _index.html_
+
+### Récupération préliminaire des données (à la main)
+
+Admettons maintenant que vous désirez tester avec des données de Caen dans le Calvados (département 14) :
+
+1. Aller sur [le site web de production](https://vitemadose.covidtracker.fr/)
+2. Ouvrir la console web du navigateur que vous utilisez
+2. Saisir _14000_ et sélectionner _14000 - Caen_
+3. Récupérer (via un onglet supposément nommé _Network_ par exemple) le contenu des fichiers _14.json_ qui aura les infos des centres de vaccination, ainsi que le fichier _creneaux-quotidiens.json_. S'assurer de prendre celui du bon département (URL de la forme _https://vitemadose.gitlab.io/vitemadose/14/creneaux-quotidiens.json_). Récupérer aussi le fichier _stats.json_ et le fichier _departements.json_ (par exemple via l'URL _https://vitemadose.gitlab.io/vitemadose/departements.json_)
```
Comment (pylapp) [inline HOW-TO.md:35]: "Récupérer **via la console du navigateur**"

File: HOW-TO.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+5. Dans le sous-dossier _vitemadose_ du dossier web, créér un sous-dossier _14_ et y placer le fichier _creneaux-quotidiens.json_
+
+Maintenant, il ne reste plus qu'à compiler le projet _Xcode_ sur un appareil et lancer l'application. L'appareil doit être sur le même réseau local que la machine ayant le serveur web.
+
+Astuce pour savoir si on affiche les données locales ou non : modifier le contenu des fichiers JSON avec des éléments exotiques comme :
+- dans _stats.json_, pour l'objet **tout_departement**, remplacer la valeur du champ "disponibles" par des nombres reconnaissables
+- dans le fichier _14.json_, changer le nom d'un centre de vaccination
+
+Si toutefois, malgré tout ceci, vous pensez récupérer des données depuis le site web de production, il est possible que la configuration à distance (via _Firebase_)  ait pris la main et écrasé la configuration par défaut renseignée dans le fichier _remote-config.plist_. Auquel cas, y aller plus brutalement en remplaçant la valeur retournée par la propriété calculée `baseUrl` dans l'extension de `RemoteConfiguration` par l'adresse IP du serveur local, comme :
```
Comment (pylapp) [inline HOW-TO.md:45]: Espace en trop

File: ViteMaDose.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -1214,14 +1218,14 @@
...
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = 9363PZ22AY;
 				INFOPLIST_FILE = ViteMaDose/Info.plist;
 				LD_RUNPATH_SEARCH_PATHS = (
 					"$(inherited)",
 					"@executable_path/Frameworks",
 				);
 				MARKETING_VERSION = 1.0;
-				PRODUCT_BUNDLE_IDENTIFIER = app.vitemadose.debug;
+				PRODUCT_BUNDLE_IDENTIFIER = pylapp.vitemadose.ios;
```
Comment (pylapp) [inline ViteMaDose.xcodeproj/project.pbxproj:1228]: Oups

File: ViteMaDose.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -1214,14 +1218,14 @@
 				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 841539;
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = 9363PZ22AY;
```
Comment (pylapp) [inline ViteMaDose.xcodeproj/project.pbxproj:1221]: Oups

File: ViteMaDose/Models/DailySlots.swift [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+// Author: Pierre-Yves LAPERSONNE et al.
+
+import Foundation
+
+// MARK: - Daily Slots For Location
+
+/// Allows to gather for a departement all the daily slots sent by the backend.
+public struct DailySlotsForDistrict: Codable {
+
+    /// Location code for the department, e.g. 31 for Haute-Garonne
```
Comment (pylapp) [inline ViteMaDose/Models/DailySlots.swift:16]: "Location code for the department, e.g. **"31"** for **"Haute-Garonne"**"

File: ViteMaDose/Models/DailySlots.swift [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+        }
+        for slot in dailySlots {
+            total += slot.numberOfBoosterShotSlots(for: centerIdentfier)
+        }
+        return total
+    }
+
+    /// Returns the slots for the given day.
+    /// - Parameter day: The date to use for filtering
+    /// - Returns: Aerray of `DailySlot`, can be empty if nothing found for this `day`
```
Comment (pylapp) [inline ViteMaDose/Models/DailySlots.swift:45]: Typo: **"Array"**

File: ViteMaDose/Models/DailySlots.swift [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+        }
+        return count
+    }
+}
+
+// MARK: - Slot
+
+public struct Slot: Codable {
+
+    /// Code for the booking platform related to the location when there are slots (e.g. "mesoigner3031" ot "doctolib203443pid88863")
```
Comment (pylapp) [inline ViteMaDose/Models/DailySlots.swift:99]: Typo: **"where"**

File: ViteMaDose/Models/DailySlots.swift [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+extension Slot {
+
+    /// - Returns Bool: True if this `Slot` contains a booster shot tag with a positive number of slots ; false otherwise
+    func hasBoosterShots() -> Bool {
+        numberOfBoosterShotsSlots() > 0
+    }
+
+    /// - Returns: The number of third shot / booster shot this slot have
```
Comment (pylapp) [inline ViteMaDose/Models/DailySlots.swift:117]: Typo: "shot**s**"

File: ViteMaDose/Models/DailySlots.swift [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+    func numberOfBoosterShotsSlots() -> Int {
+        guard let slotsPerTag = slotsPerTag,
+              let boosterShotTag = slotsPerTag.first(where: { $0.tag == .thirdShot }),
+              let slotsNumber = boosterShotTag.slotsNumber else {
+            return 0
+        }
+        return slotsNumber
+    }
+
+    /// - Returns: The global number of shots this slot have
```
Comment (pylapp) [inline ViteMaDose/Models/DailySlots.swift:127]: Typo: "**has**"

File: ViteMaDose/Models/DatedSlotsForCentres.swift [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+// MARK: - Date Slots For Centres - Booster Shots
+
+extension DatedSlotsForCentres {
+
+    /// Returns if there is some `Slot` with booster shots for the given `VaccinationCentre` in this `DatedSlotsForCentres`.
+    /// - Parameter centre: The centre to look for booster shots
+    /// - Returns: False if there is no centre stored in this `DatedSlotsForCentres` or if there is no booster shot, true if there are booster shots for this centre.
+    func hasBoosterShot(for centre: VaccinationCentre) -> Bool {
```
Comment (pylapp) [inline ViteMaDose/Models/DatedSlotsForCentres.swift:68]: Rename to "**hasBoosterShots**"

**Comment by pylapp** [comment]
Duplicated in #152, should be closed once comments here have been processed.

**Comment by pylapp** [comment]
See specific branch with copy-pasted evolutions in: https://github.com/CovidTrackerFr/vitemadose-ios/commits/feature/142_third-dose

**Pull request closed without merging**